### PR TITLE
Fix sum() forward for NJT

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #131945

This PR solves two problems with `sum()` support in NJT:
* `sum()` over a dim with `keepdim=True` returns the wrong shape (i.e. it'll keep the wrong dim). This is a long-standing bug from way back in #112519.
* Historically, we've only supported `sum()` over a dim and not a full reduction. This PR adds the full reduction form (forward only, backward still fails).

cc @cpuhrsch @bhosmer @drisspg @soulitzer @davidberard98 @YuqingJ

### CODE REVIEW

**Review by davidberard98 — APPROVED** [review]
lgtm!

File: torch/nested/_internal/ops.py [hunk 1]
```diff
@@ -941,8 +952,7 @@ def sum_dim_IntList(func, *args, **kwargs):
             )  # need to read offsets --> pad jagged dimension and apply sum
 
         if new_kwargs["keepdim"]:
-            # TODO: Fix this; it's a bug. should be unsqueezing on ragged_idx
-            out = out.unsqueeze(0)
+            out = out.unsqueeze(inp._ragged_idx)
```
Comment (jananisriram) [inline torch/nested/_internal/ops.py:955]: looks great, and happy to see that this now also works with transposed NTs!

**Review by jananisriram — APPROVED** [review]
looks great!! thanks for adding this support :)

**Comment by jbschlosser** [comment]
The reason I haven't landed this yet is because this uncovered torch.compile limitations with calculating the max seqlen metadata on the fly, a calculation that is required for any ops that utilize the jagged->padded, computation, padded->jagged fallback approach.

**Pull request closed without merging**